### PR TITLE
Update weak.rst

### PR DESCRIPTION
**Pull request by aleksandrshoshiashvili**

### CODE REVIEW

File: docs/weak.rst [hunk 1]
```diff
@@ -418,7 +418,7 @@ on top of basic runtime support.
 The first set of use cases will require more direct language support.
 To that end, I propose adding two new variable attributes,
 :code:`@weak` and :code:`@unowned`.  I also propose a small slate of
-new features which directly address the problem of capturing a value
```
Comment (harlanhaskins) [inline docs/weak.rst]: This doesn't need a comma
http://www.grammarly.com/answers/questions/17353-use-of-comma-before-which-and-but/

File: docs/weak.rst [hunk 2]
```diff
@@ -426,7 +426,7 @@ Proposed Variable Attributes
 ============================
 
 In the following discussion, a "variable-like" declaration is any
-declaration which binds a name to a (possibly mutable) value of
```
Comment (harlanhaskins) [inline docs/weak.rst]: No comma here either

File: docs/weak.rst [hunk 3]
```diff
@@ -418,7 +418,7 @@ on top of basic runtime support.
 The first set of use cases will require more direct language support.
 To that end, I propose adding two new variable attributes,
 :code:`@weak` and :code:`@unowned`.  I also propose a small slate of
-new features which directly address the problem of capturing a value
+new features, which directly address the problem of capturing a value
```
Comment (harlanhaskins) [inline docs/weak.rst]: This doesn't need a comma
http://www.grammarly.com/answers/questions/17353-use-of-comma-before-which-and-but/

File: docs/weak.rst [hunk 4]
```diff
@@ -864,7 +864,7 @@ expression which is semantically constrained to be one of:
 
 Such expressions have two useful traits:
 
-- They always end in an identifier which on some level meaningfully
+- They always end in an identifier, which on some level meaningfully
```
Comment (harlanhaskins) [inline docs/weak.rst]: > They always end in an identifier, which on some level meaningfully

The comma here should be:

> They always end in an identifier which, on some level, meaningfully

File: docs/weak.rst [hunk 5]
```diff
@@ -1049,7 +1049,7 @@ there were a reason to prefer using a single statement there).  I
 don't think it does any harm, but I wouldn't mourn it, either.  I do
 think that generalizing the initializer to an arbitrary expression
 would be a serious mistake, because readers are naturally going to
-overlook code which occurs inside the closure, and promoting side
+overlook code, which occurs inside the closure, and promoting side
```
Comment (harlanhaskins) [inline docs/weak.rst]: No comma here either.

**Comment by jrose-apple** [comment]
Actually, most of these are using "which" to mean "that" rather than the subordinate clause that needs a comma, and that is correct (American) English. I think we should leave the document as is.

**Pull request closed without merging**